mapusaurus/censusdata/views.py

```diff
@@ -1,13 +1,183 @@
+        hma = sum([entry['hma'] for entry in collector])
+        hma_pct = 1.0 * hma / peer_total
+        return {
+                'lma': lma, 
+                'lma_pct': 1.0 * lma / peer_total, 
+                'mma': mma,
+                'mma_pct': 1.0 * mma / peer_total,
+                'hma': hma,
+                'hma_pct': 1.0 * hma / peer_total,
+                'lar_total': peer_total
+                }
+    else:
+        return {
+            'lar_total': 0,
+            'lma': 0, 
+            'lma_pct': 0, 
+            'mma': 0,
+            'mma_pct': 0,
+            'hma': 0,
+            'hma_pct': 0
+            }
+def minority_aggregation_as_json(request):
+    """
+    aggregates minority population ranges and LAR counts 
+    for a lender in an MSA
+    by tract, msa and county
+    """
+    lar_data = loan_originations_as_json(request)
+    lender = Institution.objects.get(institution_id=request.GET.get('lender'))
+    tracts = Geo.objects.filter(geo_type=Geo.TRACT_TYPE, cbsa=request.GET.get('metro'))
+    metro = Geo.objects.get(geo_type=Geo.METRO_TYPE, geoid=request.GET.get('metro'))
+    lender_stats = assemble_stats(lar_data, tracts)
+    # MSA
+    msa_pop, msa_minority_ct, msa_minority_pct = tally_msa_minority_stats(tracts)
+    msa_stats = {
+        'minority_ct': msa_minority_ct,
+        'minority_pct': msa_minority_pct
+    }
+    # PEERS
+    peers = get_peer_list(lender, metro)
+    if peers:
+        peer_data_collector = []
+        for peer in peers:
+            peer_request = HttpRequest()
+            peer_request.GET['lender'] = peer.institution.institution_id
+            peer_request.GET['metro']= metro.geoid
+            peer_lar_data = loan_originations_as_json(peer_request)
+            peer_data_collector.append(assemble_stats(peer_lar_data, tracts))
+        if len(peer_data_collector) == 1:
+            peer_stats = peer_data_collector[0]
+        else:
+            peer_stats = combine_peer_stats(peer_data_collector)
+        # ODDS
+        target_mm = lender_stats['hma'] + lender_stats['mma']
+        target_non = lender_stats['lma']
+        peer_mm = peer_stats['hma'] + peer_stats['mma']
+        peer_non = peer_stats['lma']
+        odds = odds_ratio(target_mm, target_non, peer_mm, peer_non)
+    else:
+        peer_stats = None
+        odds = None
+    # BY COUNTY -- TODO
+    county_ids = sorted(set([tract.geoid[:5] for tract in tracts]))
+    county_stats = {county_id: {} for county_id in county_ids}
+    for county_id in county_ids:
+        county_tracts = Geo.objects.filter(geo_type=Geo.TRACT_TYPE, state=county_id[:2], county=county_id[2:])
+        county_stats[county_id] = assemble_stats(lar_data, county_tracts)
+
+    return {
+        'msa': msa_stats,
+        'lender': lender_stats,
+        'peers': peer_stats,
+        'odds': odds,
+        'counties': county_stats,
+        }
+
+def odds_ratio(target_mm, target_non, peer_mm, peer_non):
```

Comment:
this is based on a formula stephen kennedy provided, via makada. stephen has since indicated the algorithm request is likely to change, and suggested we leave the first iteration as-is until this is nailed down.